core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaConsumerHelper.java

```diff
@@ -79,12 +79,18 @@ public static Consumer createConsumer(String bootStrapServers, String consumerPr
         }
     }
 
-    public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer) {
+    public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer, ConsumerCommonConfigs consumerCommonConfigs) {
+        // default wait time for poll
+        Long durationForPolling = 500L;
+        // use user defined consumer.pollingTime if provided
+        if (consumerCommonConfigs != null)
+            durationForPolling = ofNullable(consumerCommonConfigs.getPollingTime()).orElse(500L);
```

Comment:
When you use ConsumerLocalConfigs with the derived configuration I assume you could use the `static getPollTime()` and skip defining a default, as there already exists one in KafkaConstants.

Reply:
@ddingel updated code to address your comments.